src/context/pool_query_context.c

```diff
@@ -1825,7 +1831,7 @@ PRESTOGRES_DEST prestogres_send_to_where(Node *node)
 	 * CREATE TABLE
 	 * CREATE TABLE ... AS SELECT
 	 */
-	if (IsA(node, SelectStmt) || IsA(node, InsertStmt) || IsA(node, CreateStmt) || IsA(node, CreateTableAsStmt))
+	if (IsA(node, SelectStmt) || IsA(node, InsertStmt) || IsA(node, CreateStmt) || IsA(node, CreateTableAsStmt) || IsA(node, DropStmt))
```

Comment:
@tomz  Nice :+1: 
Setting NULL directly prestogres_system_catalog_relcache seems leaking memory. And I found `prestogres_discard_system_catalog` function which is called at https://github.com/tomz/prestogres/blob/master/src/protocol/pool_proto_modules.c#L1638-L1645
So probably the best place to discard system catalog is right after the `if` block as following:

```
 if (IsA(node, CreateStmt))
 {
     CreateStmt *create_table_stmt = (CreateStmt *)node;
     if (create_table_stmt->relation->relpersistence)
         discard_temp_table_relcache();
     /* prestogres: discard system catalog to recreate them next time at prestogres_init_system_catalog */
     prestogres_discard_system_catalog();
 }
+ else if (IsA(node, DropStmt))
+ {
+     /* prestogres: discard system catalog to recreate them next time at prestogres_init_system_catalog */
+     prestogres_discard_system_catalog();
+ }
```